docs/async-pattern.md

```diff
@@ -66,12 +71,12 @@ You may need  an [access token](access-token.md).
 curl --request PUT \
   --url https://localhost:2746/api/v1/workflows/<NAMESPACE>/<WORKFLOWNAME>/resume
   --header 'content-type: application/json' \
-  --header "Authorization: Bearer $ARGO_TOKEN" \
+  --header "Authorization: $ARGO_TOKEN" \
```

Comment:
I actually fixed these both ~1.5 months ago in #11714. I have [several docs PRs still](https://cloud-native.slack.com/archives/C0510EUH90V/p1696442823801189?thread_ts=1694644337.627309&cid=C0510EUH90V) waiting on review and merge 😕

cc @terrytangyuan